zio/zstio/reader.go

```diff
@@ -11,12 +11,15 @@ import (
 
 func NewReader(zctx *zed.Context, r io.Reader) (*zst.Reader, error) {
 	reader, ok := r.(storage.Reader)
-	if !ok {
-		return nil, errors.New("zst must be used with a seekable input")
+	if ok {
+		if !storage.IsSeekable(reader) {
+			return nil, errors.New("zst must be used with a seekable input")
+		}
+		return zst.NewReaderFromStorageReader(zctx, reader)
 	}
-	seeker, err := storage.NewSeeker(reader)
-	if err != nil {
-		return nil, errors.New("zst must be used with a seekable input")
-	}
-	return zst.NewReaderFromSeeker(zctx, seeker)
+	// This can't be the zed system (which always using package storage)
+	// so it must be a third party using he zst library.  We could assert
+	// for io.ReaderAt and io.Seeker and use the seeker to get the size
+	// and make a zst reader this way, but for now, we just say this is not supported.
+	return nil, errors.New("zst does not yet support non-storage package implementations")
```

Comment:
This part's hard to love, but I say we merge it as is and see about cleaning it up later.

Reply:
:-)